tests/test_sqs.py

```diff
@@ -74,3 +74,114 @@ def test_sqs_set_encryption(self):
             QueueUrl=queue_url,
             AttributeNames=['All'])['Attributes']['KmsMasterKeyId']
         self.assertEqual(check_master_key, 'c4816d44-73c3-4eed-a7cc-d52a74fa3294')
+
+
+    @functional
+    def test_sqs_remove_matched(self):
+        session_factory = self.replay_flight_data('test_sqs_remove_matched')
+        client = session_factory().client('sqs')
+        name = 'test-sqs-remove-matched-1'
+        queue_url = client.create_queue(QueueName=name)['QueueUrl']
+        self.addCleanup(client.delete_queue, QueueUrl=queue_url)
+
+        client.set_queue_attributes(
+            QueueUrl=queue_url,
+            Attributes={'Policy':json.dumps({
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Sid": "SpecificAllow",
+                        "Effect": "Allow",
+                        "Principal": {
+                            "AWS": "arn:aws:iam::123456789012:root"
+                        },
+                        "Action": [
+                            "sqs:Subscribe"
+                        ]
+                    },
+                    {
+                        "Sid": "Public",
+                        "Effect": "Allow",
+                        "Principal": "*",
+                        "Action": [
+                            "sqs:GetqueueAttributes"
+                        ]
+                    }
+                ]
+            })}
+        )
+
+        p = self.load_policy({
+            'name': 'sqs-rm-matched',
+            'resource': 'sqs',
+            'filters': [
+                {'QueueUrl': queue_url},
+                {'type': 'cross-account',
+                 'whitelist': ["123456789012"]}
+            ],
+            'actions': [
+                {'type': 'remove-statements',
+                 'statement_ids': 'matched'}]
+            },
+            session_factory=session_factory)
+        resources = p.run()
+
+        self.assertEqual([r['QueueUrl'] for r in resources], [queue_url])
+
+        data = json.loads(client.get_queue_attributes(QueueUrl=resources[0]['QueueUrl'], AttributeNames=['Policy'])['Attributes']['Policy'])
+        if self.recording:
```

Comment:
this sleep is non functional, you've already loaded the data via the api, what are you sleeping for? its not going to change.